index.bs

```diff
@@ -8508,6 +8527,357 @@ worker=] algorithm:
+ <tr>
+  <td><dfn>Cookie path</dfn>
+  <td><code>path</code>
+  <td>"<code>path</code>"
+  <td>The cookie path.
+   Defaults to "<code>/</code>"
+   if omitted when [=storage.setCookie|setting a cookie=].
+ </tr>
+
+ <tr>
+  <td><dfn>Cookie domain</dfn>
+  <td><code>domain</code>
+  <td>"<code>domain</code>"
+  <td>The domain the cookie is visible to.
+ </tr>
+
+ <tr>
+  <td><dfn>Cookie secure only</dfn>
+  <td><code>secure-only-flag</code>
+  <td>"<code>secure</code>"
+  <td>Whether the cookie is a secure cookie.
+   Defaults to false if omitted when [=storage.setCookie|setting a cookie=].
+ </tr>
+
+ <tr>
+  <td><dfn>Cookie HTTP only</dfn>
+  <td><code>http-only-flag</code>
+  <td>"<code>httpOnly</code>"
+  <td>Whether the cookie is an HTTP only cookie.
+   Defaults to false if omitted when [=storage.setCookie|setting a cookie=].
+ </tr>
+
+ <tr>
+  <td><dfn>Cookie expiry time</dfn>
+  <td><code>expiry-time</code>
+  <td>"<code>expires</code>"
+  <td>When the cookie expires, specified in seconds since
+   [=Unix Epoch=]. Must not be set if omitted when
+   [=storage.setCookie|setting a cookie=].
+ </tr>
+
+ <tr>
+  <td><dfn>Cookie same site</dfn>
+  <td><code>samesite</code>
+  <td>"<code>sameSite</code>"
+  <td>Whether the cookie applies to a SameSite policy.
+    Defaults to None if omitted when [=storage.setCookie|setting a cookie=].
+    Can be set to either <code>[=Lax=]</code> or <code>[=Strict=]</code>.
+ </tr>
+</table>
+
+<div algorithm>
+To <dfn>match cookie</dfn> given |stored cookie| and |filter|:
+
+1. For each |name| → |value| in |filter|:
+  1. If |name| is "<code>value</code>":
+    1. Set |value| to [=deserialize protocol bytes=] with |value|.
+  1. Let |field name| be the field name corresponding to the JSON key |name| in
+    the [=table for cookie conversion=].
+  1. If |stored cookie|[|field name|] does not equal |value|:
+    1. Return false.
+1. Return true.
+
+</div>
+
+<div algorithm>
+To <dfn>get matching cookies</dfn> given |cookie store| and |filter|:
+
+1. Let |cookies| be a new list.
+1. For each |stored cookie| in |cookie store|:
+  1. If [=match cookie=] with |stored cookie| and |filter| is true:
+    1. Append the result of [=serialize cookie=] given |stored cookie| to |cookies|.
+1. Return |cookies|.
+
+</div>
+
+### Commands ### {#module-storage-commands}
+
+#### The storage.getCookies Command ####  {#command-storage-getCookies}
+
+The <dfn export for=commands>storage.getCookies</dfn> command retrieves zero or more [=cookies=] which [=match cookie|match=] a set of provided parameters.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+        storage.GetCookies = {
+          method: "storage.getCookies",
+          params: storage.GetCookiesParameters
+        }
+
+        storage.CookieFilter = {
+          ? ~ network.Cookie
+        }
+
+        storage.GetCookiesParameters = {
+          cookie: storage.CookieFilter,
+          ? partitionKey: storage.PartitionKey,
+        }
```

Comment:
These commands can't use [BiDi's `network.Cookie` type](https://w3c.github.io/webdriver-bidi/#type-network-Cookie) directly because in these contexts, many of the entries are optional. In the interest of limiting duplication, I've attempted to specify the `storage.CookieFilter` and `storage.PartialCookie` types in terms of `network.Cookie`, but I'm not certain the CDDL syntax I've applied is appropriate. (You can find more detail in [the message I sent to the CBOR mailing list](https://mailarchive.ietf.org/arch/msg/cbor/41_rLuyFj8upo2wJFksagTOa-HY/).)